// @aws-sdk/client-elasticache/dist-es/models/models_0.js
import { __assign } from "tslib";
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var AddTagsToResourceMessage;
(function (AddTagsToResourceMessage) {
    AddTagsToResourceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsToResourceMessage || (AddTagsToResourceMessage = {}));
export var CacheClusterNotFoundFault;
(function (CacheClusterNotFoundFault) {
    CacheClusterNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheClusterNotFoundFault || (CacheClusterNotFoundFault = {}));
export var CacheParameterGroupNotFoundFault;
(function (CacheParameterGroupNotFoundFault) {
    CacheParameterGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupNotFoundFault || (CacheParameterGroupNotFoundFault = {}));
export var CacheSecurityGroupNotFoundFault;
(function (CacheSecurityGroupNotFoundFault) {
    CacheSecurityGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroupNotFoundFault || (CacheSecurityGroupNotFoundFault = {}));
export var CacheSubnetGroupNotFoundFault;
(function (CacheSubnetGroupNotFoundFault) {
    CacheSubnetGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroupNotFoundFault || (CacheSubnetGroupNotFoundFault = {}));
export var InvalidARNFault;
(function (InvalidARNFault) {
    InvalidARNFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidARNFault || (InvalidARNFault = {}));
export var InvalidReplicationGroupStateFault;
(function (InvalidReplicationGroupStateFault) {
    InvalidReplicationGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidReplicationGroupStateFault || (InvalidReplicationGroupStateFault = {}));
export var ReplicationGroupNotFoundFault;
(function (ReplicationGroupNotFoundFault) {
    ReplicationGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupNotFoundFault || (ReplicationGroupNotFoundFault = {}));
export var ReservedCacheNodeNotFoundFault;
(function (ReservedCacheNodeNotFoundFault) {
    ReservedCacheNodeNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodeNotFoundFault || (ReservedCacheNodeNotFoundFault = {}));
export var SnapshotNotFoundFault;
(function (SnapshotNotFoundFault) {
    SnapshotNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotNotFoundFault || (SnapshotNotFoundFault = {}));
export var TagListMessage;
(function (TagListMessage) {
    TagListMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagListMessage || (TagListMessage = {}));
export var TagQuotaPerResourceExceeded;
(function (TagQuotaPerResourceExceeded) {
    TagQuotaPerResourceExceeded.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagQuotaPerResourceExceeded || (TagQuotaPerResourceExceeded = {}));
export var UserGroupNotFoundFault;
(function (UserGroupNotFoundFault) {
    UserGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroupNotFoundFault || (UserGroupNotFoundFault = {}));
export var UserNotFoundFault;
(function (UserNotFoundFault) {
    UserNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserNotFoundFault || (UserNotFoundFault = {}));
export var AllowedNodeTypeModificationsMessage;
(function (AllowedNodeTypeModificationsMessage) {
    AllowedNodeTypeModificationsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AllowedNodeTypeModificationsMessage || (AllowedNodeTypeModificationsMessage = {}));
export var AuthorizationAlreadyExistsFault;
(function (AuthorizationAlreadyExistsFault) {
    AuthorizationAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizationAlreadyExistsFault || (AuthorizationAlreadyExistsFault = {}));
export var AuthorizeCacheSecurityGroupIngressMessage;
(function (AuthorizeCacheSecurityGroupIngressMessage) {
    AuthorizeCacheSecurityGroupIngressMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeCacheSecurityGroupIngressMessage || (AuthorizeCacheSecurityGroupIngressMessage = {}));
export var EC2SecurityGroup;
(function (EC2SecurityGroup) {
    EC2SecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EC2SecurityGroup || (EC2SecurityGroup = {}));
export var CacheSecurityGroup;
(function (CacheSecurityGroup) {
    CacheSecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroup || (CacheSecurityGroup = {}));
export var AuthorizeCacheSecurityGroupIngressResult;
(function (AuthorizeCacheSecurityGroupIngressResult) {
    AuthorizeCacheSecurityGroupIngressResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeCacheSecurityGroupIngressResult || (AuthorizeCacheSecurityGroupIngressResult = {}));
export var InvalidCacheSecurityGroupStateFault;
(function (InvalidCacheSecurityGroupStateFault) {
    InvalidCacheSecurityGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidCacheSecurityGroupStateFault || (InvalidCacheSecurityGroupStateFault = {}));
export var InvalidParameterCombinationException;
(function (InvalidParameterCombinationException) {
    InvalidParameterCombinationException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidParameterCombinationException || (InvalidParameterCombinationException = {}));
export var InvalidParameterValueException;
(function (InvalidParameterValueException) {
    InvalidParameterValueException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidParameterValueException || (InvalidParameterValueException = {}));
export var BatchApplyUpdateActionMessage;
(function (BatchApplyUpdateActionMessage) {
    BatchApplyUpdateActionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchApplyUpdateActionMessage || (BatchApplyUpdateActionMessage = {}));
export var ServiceUpdateNotFoundFault;
(function (ServiceUpdateNotFoundFault) {
    ServiceUpdateNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceUpdateNotFoundFault || (ServiceUpdateNotFoundFault = {}));
export var UpdateActionStatus;
(function (UpdateActionStatus) {
    UpdateActionStatus["COMPLETE"] = "complete";
    UpdateActionStatus["IN_PROGRESS"] = "in-progress";
    UpdateActionStatus["NOT_APPLICABLE"] = "not-applicable";
    UpdateActionStatus["NOT_APPLIED"] = "not-applied";
    UpdateActionStatus["SCHEDULED"] = "scheduled";
    UpdateActionStatus["SCHEDULING"] = "scheduling";
    UpdateActionStatus["STOPPED"] = "stopped";
    UpdateActionStatus["STOPPING"] = "stopping";
    UpdateActionStatus["WAITING_TO_START"] = "waiting-to-start";
})(UpdateActionStatus || (UpdateActionStatus = {}));
export var ProcessedUpdateAction;
(function (ProcessedUpdateAction) {
    ProcessedUpdateAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProcessedUpdateAction || (ProcessedUpdateAction = {}));
export var UnprocessedUpdateAction;
(function (UnprocessedUpdateAction) {
    UnprocessedUpdateAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UnprocessedUpdateAction || (UnprocessedUpdateAction = {}));
export var UpdateActionResultsMessage;
(function (UpdateActionResultsMessage) {
    UpdateActionResultsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateActionResultsMessage || (UpdateActionResultsMessage = {}));
export var BatchStopUpdateActionMessage;
(function (BatchStopUpdateActionMessage) {
    BatchStopUpdateActionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchStopUpdateActionMessage || (BatchStopUpdateActionMessage = {}));
export var CompleteMigrationMessage;
(function (CompleteMigrationMessage) {
    CompleteMigrationMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompleteMigrationMessage || (CompleteMigrationMessage = {}));
export var AutomaticFailoverStatus;
(function (AutomaticFailoverStatus) {
    AutomaticFailoverStatus["DISABLED"] = "disabled";
    AutomaticFailoverStatus["DISABLING"] = "disabling";
    AutomaticFailoverStatus["ENABLED"] = "enabled";
    AutomaticFailoverStatus["ENABLING"] = "enabling";
})(AutomaticFailoverStatus || (AutomaticFailoverStatus = {}));
export var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Endpoint || (Endpoint = {}));
export var GlobalReplicationGroupInfo;
(function (GlobalReplicationGroupInfo) {
    GlobalReplicationGroupInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalReplicationGroupInfo || (GlobalReplicationGroupInfo = {}));
export var CloudWatchLogsDestinationDetails;
(function (CloudWatchLogsDestinationDetails) {
    CloudWatchLogsDestinationDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CloudWatchLogsDestinationDetails || (CloudWatchLogsDestinationDetails = {}));
export var KinesisFirehoseDestinationDetails;
(function (KinesisFirehoseDestinationDetails) {
    KinesisFirehoseDestinationDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisFirehoseDestinationDetails || (KinesisFirehoseDestinationDetails = {}));
export var DestinationDetails;
(function (DestinationDetails) {
    DestinationDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DestinationDetails || (DestinationDetails = {}));
export var DestinationType;
(function (DestinationType) {
    DestinationType["CloudWatchLogs"] = "cloudwatch-logs";
    DestinationType["KinesisFirehose"] = "kinesis-firehose";
})(DestinationType || (DestinationType = {}));
export var LogFormat;
(function (LogFormat) {
    LogFormat["JSON"] = "json";
    LogFormat["TEXT"] = "text";
})(LogFormat || (LogFormat = {}));
export var LogType;
(function (LogType) {
    LogType["SLOW_LOG"] = "slow-log";
})(LogType || (LogType = {}));
export var LogDeliveryConfigurationStatus;
(function (LogDeliveryConfigurationStatus) {
    LogDeliveryConfigurationStatus["ACTIVE"] = "active";
    LogDeliveryConfigurationStatus["DISABLING"] = "disabling";
    LogDeliveryConfigurationStatus["ENABLING"] = "enabling";
    LogDeliveryConfigurationStatus["ERROR"] = "error";
    LogDeliveryConfigurationStatus["MODIFYING"] = "modifying";
})(LogDeliveryConfigurationStatus || (LogDeliveryConfigurationStatus = {}));
export var LogDeliveryConfiguration;
(function (LogDeliveryConfiguration) {
    LogDeliveryConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LogDeliveryConfiguration || (LogDeliveryConfiguration = {}));
export var MultiAZStatus;
(function (MultiAZStatus) {
    MultiAZStatus["DISABLED"] = "disabled";
    MultiAZStatus["ENABLED"] = "enabled";
})(MultiAZStatus || (MultiAZStatus = {}));
export var NodeGroupMember;
(function (NodeGroupMember) {
    NodeGroupMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupMember || (NodeGroupMember = {}));
export var NodeGroup;
(function (NodeGroup) {
    NodeGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroup || (NodeGroup = {}));
export var PendingAutomaticFailoverStatus;
(function (PendingAutomaticFailoverStatus) {
    PendingAutomaticFailoverStatus["DISABLED"] = "disabled";
    PendingAutomaticFailoverStatus["ENABLED"] = "enabled";
})(PendingAutomaticFailoverStatus || (PendingAutomaticFailoverStatus = {}));
export var PendingLogDeliveryConfiguration;
(function (PendingLogDeliveryConfiguration) {
    PendingLogDeliveryConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingLogDeliveryConfiguration || (PendingLogDeliveryConfiguration = {}));
export var SlotMigration;
(function (SlotMigration) {
    SlotMigration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SlotMigration || (SlotMigration = {}));
export var ReshardingStatus;
(function (ReshardingStatus) {
    ReshardingStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReshardingStatus || (ReshardingStatus = {}));
export var UserGroupsUpdateStatus;
(function (UserGroupsUpdateStatus) {
    UserGroupsUpdateStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroupsUpdateStatus || (UserGroupsUpdateStatus = {}));
export var ReplicationGroupPendingModifiedValues;
(function (ReplicationGroupPendingModifiedValues) {
    ReplicationGroupPendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupPendingModifiedValues || (ReplicationGroupPendingModifiedValues = {}));
export var ReplicationGroup;
(function (ReplicationGroup) {
    ReplicationGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroup || (ReplicationGroup = {}));
export var CompleteMigrationResponse;
(function (CompleteMigrationResponse) {
    CompleteMigrationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompleteMigrationResponse || (CompleteMigrationResponse = {}));
export var ReplicationGroupNotUnderMigrationFault;
(function (ReplicationGroupNotUnderMigrationFault) {
    ReplicationGroupNotUnderMigrationFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupNotUnderMigrationFault || (ReplicationGroupNotUnderMigrationFault = {}));
export var CopySnapshotMessage;
(function (CopySnapshotMessage) {
    CopySnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopySnapshotMessage || (CopySnapshotMessage = {}));
export var NodeGroupConfiguration;
(function (NodeGroupConfiguration) {
    NodeGroupConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupConfiguration || (NodeGroupConfiguration = {}));
export var NodeSnapshot;
(function (NodeSnapshot) {
    NodeSnapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeSnapshot || (NodeSnapshot = {}));
export var Snapshot;
(function (Snapshot) {
    Snapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Snapshot || (Snapshot = {}));
export var CopySnapshotResult;
(function (CopySnapshotResult) {
    CopySnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopySnapshotResult || (CopySnapshotResult = {}));
export var InvalidSnapshotStateFault;
(function (InvalidSnapshotStateFault) {
    InvalidSnapshotStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSnapshotStateFault || (InvalidSnapshotStateFault = {}));
export var SnapshotAlreadyExistsFault;
(function (SnapshotAlreadyExistsFault) {
    SnapshotAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotAlreadyExistsFault || (SnapshotAlreadyExistsFault = {}));
export var SnapshotQuotaExceededFault;
(function (SnapshotQuotaExceededFault) {
    SnapshotQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotQuotaExceededFault || (SnapshotQuotaExceededFault = {}));
export var CacheClusterAlreadyExistsFault;
(function (CacheClusterAlreadyExistsFault) {
    CacheClusterAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheClusterAlreadyExistsFault || (CacheClusterAlreadyExistsFault = {}));
export var ClusterQuotaForCustomerExceededFault;
(function (ClusterQuotaForCustomerExceededFault) {
    ClusterQuotaForCustomerExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ClusterQuotaForCustomerExceededFault || (ClusterQuotaForCustomerExceededFault = {}));
export var AZMode;
(function (AZMode) {
    AZMode["CROSS_AZ"] = "cross-az";
    AZMode["SINGLE_AZ"] = "single-az";
})(AZMode || (AZMode = {}));
export var LogDeliveryConfigurationRequest;
(function (LogDeliveryConfigurationRequest) {
    LogDeliveryConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LogDeliveryConfigurationRequest || (LogDeliveryConfigurationRequest = {}));
export var OutpostMode;
(function (OutpostMode) {
    OutpostMode["CROSS_OUTPOST"] = "cross-outpost";
    OutpostMode["SINGLE_OUTPOST"] = "single-outpost";
})(OutpostMode || (OutpostMode = {}));
export var CreateCacheClusterMessage;
(function (CreateCacheClusterMessage) {
    CreateCacheClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheClusterMessage || (CreateCacheClusterMessage = {}));
export var CacheNode;
(function (CacheNode) {
    CacheNode.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheNode || (CacheNode = {}));
export var CacheParameterGroupStatus;
(function (CacheParameterGroupStatus) {
    CacheParameterGroupStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupStatus || (CacheParameterGroupStatus = {}));
export var CacheSecurityGroupMembership;
(function (CacheSecurityGroupMembership) {
    CacheSecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroupMembership || (CacheSecurityGroupMembership = {}));
export var NotificationConfiguration;
(function (NotificationConfiguration) {
    NotificationConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NotificationConfiguration || (NotificationConfiguration = {}));
export var PendingModifiedValues;
(function (PendingModifiedValues) {
    PendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingModifiedValues || (PendingModifiedValues = {}));
export var SecurityGroupMembership;
(function (SecurityGroupMembership) {
    SecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityGroupMembership || (SecurityGroupMembership = {}));
export var CacheCluster;
(function (CacheCluster) {
    CacheCluster.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheCluster || (CacheCluster = {}));
export var CreateCacheClusterResult;
(function (CreateCacheClusterResult) {
    CreateCacheClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheClusterResult || (CreateCacheClusterResult = {}));
export var InsufficientCacheClusterCapacityFault;
(function (InsufficientCacheClusterCapacityFault) {
    InsufficientCacheClusterCapacityFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsufficientCacheClusterCapacityFault || (InsufficientCacheClusterCapacityFault = {}));
export var InvalidVPCNetworkStateFault;
(function (InvalidVPCNetworkStateFault) {
    InvalidVPCNetworkStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidVPCNetworkStateFault || (InvalidVPCNetworkStateFault = {}));
export var NodeQuotaForClusterExceededFault;
(function (NodeQuotaForClusterExceededFault) {
    NodeQuotaForClusterExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeQuotaForClusterExceededFault || (NodeQuotaForClusterExceededFault = {}));
export var NodeQuotaForCustomerExceededFault;
(function (NodeQuotaForCustomerExceededFault) {
    NodeQuotaForCustomerExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeQuotaForCustomerExceededFault || (NodeQuotaForCustomerExceededFault = {}));
export var CacheParameterGroupAlreadyExistsFault;
(function (CacheParameterGroupAlreadyExistsFault) {
    CacheParameterGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupAlreadyExistsFault || (CacheParameterGroupAlreadyExistsFault = {}));
export var CacheParameterGroupQuotaExceededFault;
(function (CacheParameterGroupQuotaExceededFault) {
    CacheParameterGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupQuotaExceededFault || (CacheParameterGroupQuotaExceededFault = {}));
export var CreateCacheParameterGroupMessage;
(function (CreateCacheParameterGroupMessage) {
    CreateCacheParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheParameterGroupMessage || (CreateCacheParameterGroupMessage = {}));
export var CacheParameterGroup;
(function (CacheParameterGroup) {
    CacheParameterGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroup || (CacheParameterGroup = {}));
export var CreateCacheParameterGroupResult;
(function (CreateCacheParameterGroupResult) {
    CreateCacheParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheParameterGroupResult || (CreateCacheParameterGroupResult = {}));
export var InvalidCacheParameterGroupStateFault;
(function (InvalidCacheParameterGroupStateFault) {
    InvalidCacheParameterGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidCacheParameterGroupStateFault || (InvalidCacheParameterGroupStateFault = {}));
export var CacheSecurityGroupAlreadyExistsFault;
(function (CacheSecurityGroupAlreadyExistsFault) {
    CacheSecurityGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroupAlreadyExistsFault || (CacheSecurityGroupAlreadyExistsFault = {}));
export var CacheSecurityGroupQuotaExceededFault;
(function (CacheSecurityGroupQuotaExceededFault) {
    CacheSecurityGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroupQuotaExceededFault || (CacheSecurityGroupQuotaExceededFault = {}));
export var CreateCacheSecurityGroupMessage;
(function (CreateCacheSecurityGroupMessage) {
    CreateCacheSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheSecurityGroupMessage || (CreateCacheSecurityGroupMessage = {}));
export var CreateCacheSecurityGroupResult;
(function (CreateCacheSecurityGroupResult) {
    CreateCacheSecurityGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheSecurityGroupResult || (CreateCacheSecurityGroupResult = {}));
export var CacheSubnetGroupAlreadyExistsFault;
(function (CacheSubnetGroupAlreadyExistsFault) {
    CacheSubnetGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroupAlreadyExistsFault || (CacheSubnetGroupAlreadyExistsFault = {}));
export var CacheSubnetGroupQuotaExceededFault;
(function (CacheSubnetGroupQuotaExceededFault) {
    CacheSubnetGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroupQuotaExceededFault || (CacheSubnetGroupQuotaExceededFault = {}));
export var CacheSubnetQuotaExceededFault;
(function (CacheSubnetQuotaExceededFault) {
    CacheSubnetQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetQuotaExceededFault || (CacheSubnetQuotaExceededFault = {}));
export var CreateCacheSubnetGroupMessage;
(function (CreateCacheSubnetGroupMessage) {
    CreateCacheSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheSubnetGroupMessage || (CreateCacheSubnetGroupMessage = {}));
export var AvailabilityZone;
(function (AvailabilityZone) {
    AvailabilityZone.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AvailabilityZone || (AvailabilityZone = {}));
export var SubnetOutpost;
(function (SubnetOutpost) {
    SubnetOutpost.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubnetOutpost || (SubnetOutpost = {}));
export var Subnet;
(function (Subnet) {
    Subnet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Subnet || (Subnet = {}));
export var CacheSubnetGroup;
(function (CacheSubnetGroup) {
    CacheSubnetGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroup || (CacheSubnetGroup = {}));
export var CreateCacheSubnetGroupResult;
(function (CreateCacheSubnetGroupResult) {
    CreateCacheSubnetGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCacheSubnetGroupResult || (CreateCacheSubnetGroupResult = {}));
export var InvalidSubnet;
(function (InvalidSubnet) {
    InvalidSubnet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSubnet || (InvalidSubnet = {}));
export var SubnetNotAllowedFault;
(function (SubnetNotAllowedFault) {
    SubnetNotAllowedFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubnetNotAllowedFault || (SubnetNotAllowedFault = {}));
export var CreateGlobalReplicationGroupMessage;
(function (CreateGlobalReplicationGroupMessage) {
    CreateGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalReplicationGroupMessage || (CreateGlobalReplicationGroupMessage = {}));
export var GlobalNodeGroup;
(function (GlobalNodeGroup) {
    GlobalNodeGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalNodeGroup || (GlobalNodeGroup = {}));
export var GlobalReplicationGroupMember;
(function (GlobalReplicationGroupMember) {
    GlobalReplicationGroupMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalReplicationGroupMember || (GlobalReplicationGroupMember = {}));
export var GlobalReplicationGroup;
(function (GlobalReplicationGroup) {
    GlobalReplicationGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalReplicationGroup || (GlobalReplicationGroup = {}));
export var CreateGlobalReplicationGroupResult;
(function (CreateGlobalReplicationGroupResult) {
    CreateGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalReplicationGroupResult || (CreateGlobalReplicationGroupResult = {}));
export var GlobalReplicationGroupAlreadyExistsFault;
(function (GlobalReplicationGroupAlreadyExistsFault) {
    GlobalReplicationGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalReplicationGroupAlreadyExistsFault || (GlobalReplicationGroupAlreadyExistsFault = {}));
export var ServiceLinkedRoleNotFoundFault;
(function (ServiceLinkedRoleNotFoundFault) {
    ServiceLinkedRoleNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceLinkedRoleNotFoundFault || (ServiceLinkedRoleNotFoundFault = {}));
export var CreateReplicationGroupMessage;
(function (CreateReplicationGroupMessage) {
    CreateReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateReplicationGroupMessage || (CreateReplicationGroupMessage = {}));
export var CreateReplicationGroupResult;
(function (CreateReplicationGroupResult) {
    CreateReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateReplicationGroupResult || (CreateReplicationGroupResult = {}));
export var GlobalReplicationGroupNotFoundFault;
(function (GlobalReplicationGroupNotFoundFault) {
    GlobalReplicationGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalReplicationGroupNotFoundFault || (GlobalReplicationGroupNotFoundFault = {}));
export var InvalidCacheClusterStateFault;
(function (InvalidCacheClusterStateFault) {
    InvalidCacheClusterStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidCacheClusterStateFault || (InvalidCacheClusterStateFault = {}));
export var InvalidGlobalReplicationGroupStateFault;
(function (InvalidGlobalReplicationGroupStateFault) {
    InvalidGlobalReplicationGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidGlobalReplicationGroupStateFault || (InvalidGlobalReplicationGroupStateFault = {}));
export var InvalidUserGroupStateFault;
(function (InvalidUserGroupStateFault) {
    InvalidUserGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidUserGroupStateFault || (InvalidUserGroupStateFault = {}));
export var NodeGroupsPerReplicationGroupQuotaExceededFault;
(function (NodeGroupsPerReplicationGroupQuotaExceededFault) {
    NodeGroupsPerReplicationGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupsPerReplicationGroupQuotaExceededFault || (NodeGroupsPerReplicationGroupQuotaExceededFault = {}));
export var ReplicationGroupAlreadyExistsFault;
(function (ReplicationGroupAlreadyExistsFault) {
    ReplicationGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupAlreadyExistsFault || (ReplicationGroupAlreadyExistsFault = {}));
export var CreateSnapshotMessage;
(function (CreateSnapshotMessage) {
    CreateSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSnapshotMessage || (CreateSnapshotMessage = {}));
export var CreateSnapshotResult;
(function (CreateSnapshotResult) {
    CreateSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSnapshotResult || (CreateSnapshotResult = {}));
export var SnapshotFeatureNotSupportedFault;
(function (SnapshotFeatureNotSupportedFault) {
    SnapshotFeatureNotSupportedFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotFeatureNotSupportedFault || (SnapshotFeatureNotSupportedFault = {}));
export var CreateUserMessage;
(function (CreateUserMessage) {
    CreateUserMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserMessage || (CreateUserMessage = {}));
export var DuplicateUserNameFault;
(function (DuplicateUserNameFault) {
    DuplicateUserNameFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicateUserNameFault || (DuplicateUserNameFault = {}));
export var AuthenticationType;
(function (AuthenticationType) {
    AuthenticationType["NO_PASSWORD"] = "no-password";
    AuthenticationType["PASSWORD"] = "password";
})(AuthenticationType || (AuthenticationType = {}));
export var Authentication;
(function (Authentication) {
    Authentication.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Authentication || (Authentication = {}));
export var User;
(function (User) {
    User.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(User || (User = {}));
export var UserAlreadyExistsFault;
(function (UserAlreadyExistsFault) {
    UserAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserAlreadyExistsFault || (UserAlreadyExistsFault = {}));
export var UserQuotaExceededFault;
(function (UserQuotaExceededFault) {
    UserQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserQuotaExceededFault || (UserQuotaExceededFault = {}));
export var CreateUserGroupMessage;
(function (CreateUserGroupMessage) {
    CreateUserGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserGroupMessage || (CreateUserGroupMessage = {}));
export var DefaultUserRequired;
(function (DefaultUserRequired) {
    DefaultUserRequired.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DefaultUserRequired || (DefaultUserRequired = {}));
export var UserGroupPendingChanges;
(function (UserGroupPendingChanges) {
    UserGroupPendingChanges.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroupPendingChanges || (UserGroupPendingChanges = {}));
export var UserGroup;
(function (UserGroup) {
    UserGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroup || (UserGroup = {}));
export var UserGroupAlreadyExistsFault;
(function (UserGroupAlreadyExistsFault) {
    UserGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroupAlreadyExistsFault || (UserGroupAlreadyExistsFault = {}));
export var UserGroupQuotaExceededFault;
(function (UserGroupQuotaExceededFault) {
    UserGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserGroupQuotaExceededFault || (UserGroupQuotaExceededFault = {}));
export var DecreaseNodeGroupsInGlobalReplicationGroupMessage;
(function (DecreaseNodeGroupsInGlobalReplicationGroupMessage) {
    DecreaseNodeGroupsInGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DecreaseNodeGroupsInGlobalReplicationGroupMessage || (DecreaseNodeGroupsInGlobalReplicationGroupMessage = {}));
export var DecreaseNodeGroupsInGlobalReplicationGroupResult;
(function (DecreaseNodeGroupsInGlobalReplicationGroupResult) {
    DecreaseNodeGroupsInGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DecreaseNodeGroupsInGlobalReplicationGroupResult || (DecreaseNodeGroupsInGlobalReplicationGroupResult = {}));
export var ConfigureShard;
(function (ConfigureShard) {
    ConfigureShard.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConfigureShard || (ConfigureShard = {}));
export var DecreaseReplicaCountMessage;
(function (DecreaseReplicaCountMessage) {
    DecreaseReplicaCountMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DecreaseReplicaCountMessage || (DecreaseReplicaCountMessage = {}));
export var DecreaseReplicaCountResult;
(function (DecreaseReplicaCountResult) {
    DecreaseReplicaCountResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DecreaseReplicaCountResult || (DecreaseReplicaCountResult = {}));
export var NoOperationFault;
(function (NoOperationFault) {
    NoOperationFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NoOperationFault || (NoOperationFault = {}));
export var DeleteCacheClusterMessage;
(function (DeleteCacheClusterMessage) {
    DeleteCacheClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCacheClusterMessage || (DeleteCacheClusterMessage = {}));
export var DeleteCacheClusterResult;
(function (DeleteCacheClusterResult) {
    DeleteCacheClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCacheClusterResult || (DeleteCacheClusterResult = {}));
export var DeleteCacheParameterGroupMessage;
(function (DeleteCacheParameterGroupMessage) {
    DeleteCacheParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCacheParameterGroupMessage || (DeleteCacheParameterGroupMessage = {}));
export var DeleteCacheSecurityGroupMessage;
(function (DeleteCacheSecurityGroupMessage) {
    DeleteCacheSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCacheSecurityGroupMessage || (DeleteCacheSecurityGroupMessage = {}));
export var CacheSubnetGroupInUse;
(function (CacheSubnetGroupInUse) {
    CacheSubnetGroupInUse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroupInUse || (CacheSubnetGroupInUse = {}));
export var DeleteCacheSubnetGroupMessage;
(function (DeleteCacheSubnetGroupMessage) {
    DeleteCacheSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCacheSubnetGroupMessage || (DeleteCacheSubnetGroupMessage = {}));
export var DeleteGlobalReplicationGroupMessage;
(function (DeleteGlobalReplicationGroupMessage) {
    DeleteGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalReplicationGroupMessage || (DeleteGlobalReplicationGroupMessage = {}));
export var DeleteGlobalReplicationGroupResult;
(function (DeleteGlobalReplicationGroupResult) {
    DeleteGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalReplicationGroupResult || (DeleteGlobalReplicationGroupResult = {}));
export var DeleteReplicationGroupMessage;
(function (DeleteReplicationGroupMessage) {
    DeleteReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteReplicationGroupMessage || (DeleteReplicationGroupMessage = {}));
export var DeleteReplicationGroupResult;
(function (DeleteReplicationGroupResult) {
    DeleteReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteReplicationGroupResult || (DeleteReplicationGroupResult = {}));
export var DeleteSnapshotMessage;
(function (DeleteSnapshotMessage) {
    DeleteSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSnapshotMessage || (DeleteSnapshotMessage = {}));
export var DeleteSnapshotResult;
(function (DeleteSnapshotResult) {
    DeleteSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSnapshotResult || (DeleteSnapshotResult = {}));
export var DefaultUserAssociatedToUserGroupFault;
(function (DefaultUserAssociatedToUserGroupFault) {
    DefaultUserAssociatedToUserGroupFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DefaultUserAssociatedToUserGroupFault || (DefaultUserAssociatedToUserGroupFault = {}));
export var DeleteUserMessage;
(function (DeleteUserMessage) {
    DeleteUserMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteUserMessage || (DeleteUserMessage = {}));
export var InvalidUserStateFault;
(function (InvalidUserStateFault) {
    InvalidUserStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidUserStateFault || (InvalidUserStateFault = {}));
export var DeleteUserGroupMessage;
(function (DeleteUserGroupMessage) {
    DeleteUserGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteUserGroupMessage || (DeleteUserGroupMessage = {}));
export var CacheClusterMessage;
(function (CacheClusterMessage) {
    CacheClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheClusterMessage || (CacheClusterMessage = {}));
export var DescribeCacheClustersMessage;
(function (DescribeCacheClustersMessage) {
    DescribeCacheClustersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheClustersMessage || (DescribeCacheClustersMessage = {}));
export var CacheEngineVersion;
(function (CacheEngineVersion) {
    CacheEngineVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheEngineVersion || (CacheEngineVersion = {}));
export var CacheEngineVersionMessage;
(function (CacheEngineVersionMessage) {
    CacheEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheEngineVersionMessage || (CacheEngineVersionMessage = {}));
export var DescribeCacheEngineVersionsMessage;
(function (DescribeCacheEngineVersionsMessage) {
    DescribeCacheEngineVersionsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheEngineVersionsMessage || (DescribeCacheEngineVersionsMessage = {}));
export var CacheParameterGroupsMessage;
(function (CacheParameterGroupsMessage) {
    CacheParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupsMessage || (CacheParameterGroupsMessage = {}));
export var DescribeCacheParameterGroupsMessage;
(function (DescribeCacheParameterGroupsMessage) {
    DescribeCacheParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheParameterGroupsMessage || (DescribeCacheParameterGroupsMessage = {}));
export var CacheNodeTypeSpecificValue;
(function (CacheNodeTypeSpecificValue) {
    CacheNodeTypeSpecificValue.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheNodeTypeSpecificValue || (CacheNodeTypeSpecificValue = {}));
export var CacheNodeTypeSpecificParameter;
(function (CacheNodeTypeSpecificParameter) {
    CacheNodeTypeSpecificParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheNodeTypeSpecificParameter || (CacheNodeTypeSpecificParameter = {}));
export var Parameter;
(function (Parameter) {
    Parameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Parameter || (Parameter = {}));
export var CacheParameterGroupDetails;
(function (CacheParameterGroupDetails) {
    CacheParameterGroupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupDetails || (CacheParameterGroupDetails = {}));
export var DescribeCacheParametersMessage;
(function (DescribeCacheParametersMessage) {
    DescribeCacheParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheParametersMessage || (DescribeCacheParametersMessage = {}));
export var CacheSecurityGroupMessage;
(function (CacheSecurityGroupMessage) {
    CacheSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSecurityGroupMessage || (CacheSecurityGroupMessage = {}));
export var DescribeCacheSecurityGroupsMessage;
(function (DescribeCacheSecurityGroupsMessage) {
    DescribeCacheSecurityGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheSecurityGroupsMessage || (DescribeCacheSecurityGroupsMessage = {}));
export var CacheSubnetGroupMessage;
(function (CacheSubnetGroupMessage) {
    CacheSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheSubnetGroupMessage || (CacheSubnetGroupMessage = {}));
export var DescribeCacheSubnetGroupsMessage;
(function (DescribeCacheSubnetGroupsMessage) {
    DescribeCacheSubnetGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCacheSubnetGroupsMessage || (DescribeCacheSubnetGroupsMessage = {}));
export var DescribeEngineDefaultParametersMessage;
(function (DescribeEngineDefaultParametersMessage) {
    DescribeEngineDefaultParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEngineDefaultParametersMessage || (DescribeEngineDefaultParametersMessage = {}));
export var EngineDefaults;
(function (EngineDefaults) {
    EngineDefaults.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EngineDefaults || (EngineDefaults = {}));
export var DescribeEngineDefaultParametersResult;
(function (DescribeEngineDefaultParametersResult) {
    DescribeEngineDefaultParametersResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEngineDefaultParametersResult || (DescribeEngineDefaultParametersResult = {}));
export var DescribeEventsMessage;
(function (DescribeEventsMessage) {
    DescribeEventsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventsMessage || (DescribeEventsMessage = {}));
export var Event;
(function (Event) {
    Event.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Event || (Event = {}));
export var EventsMessage;
(function (EventsMessage) {
    EventsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventsMessage || (EventsMessage = {}));
export var DescribeGlobalReplicationGroupsMessage;
(function (DescribeGlobalReplicationGroupsMessage) {
    DescribeGlobalReplicationGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalReplicationGroupsMessage || (DescribeGlobalReplicationGroupsMessage = {}));
export var DescribeGlobalReplicationGroupsResult;
(function (DescribeGlobalReplicationGroupsResult) {
    DescribeGlobalReplicationGroupsResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalReplicationGroupsResult || (DescribeGlobalReplicationGroupsResult = {}));
export var DescribeReplicationGroupsMessage;
(function (DescribeReplicationGroupsMessage) {
    DescribeReplicationGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReplicationGroupsMessage || (DescribeReplicationGroupsMessage = {}));
export var ReplicationGroupMessage;
(function (ReplicationGroupMessage) {
    ReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupMessage || (ReplicationGroupMessage = {}));
export var DescribeReservedCacheNodesMessage;
(function (DescribeReservedCacheNodesMessage) {
    DescribeReservedCacheNodesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedCacheNodesMessage || (DescribeReservedCacheNodesMessage = {}));
export var RecurringCharge;
(function (RecurringCharge) {
    RecurringCharge.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RecurringCharge || (RecurringCharge = {}));
export var ReservedCacheNode;
(function (ReservedCacheNode) {
    ReservedCacheNode.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNode || (ReservedCacheNode = {}));
export var ReservedCacheNodeMessage;
(function (ReservedCacheNodeMessage) {
    ReservedCacheNodeMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodeMessage || (ReservedCacheNodeMessage = {}));
export var DescribeReservedCacheNodesOfferingsMessage;
(function (DescribeReservedCacheNodesOfferingsMessage) {
    DescribeReservedCacheNodesOfferingsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedCacheNodesOfferingsMessage || (DescribeReservedCacheNodesOfferingsMessage = {}));
export var ReservedCacheNodesOffering;
(function (ReservedCacheNodesOffering) {
    ReservedCacheNodesOffering.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodesOffering || (ReservedCacheNodesOffering = {}));
export var ReservedCacheNodesOfferingMessage;
(function (ReservedCacheNodesOfferingMessage) {
    ReservedCacheNodesOfferingMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodesOfferingMessage || (ReservedCacheNodesOfferingMessage = {}));
export var ReservedCacheNodesOfferingNotFoundFault;
(function (ReservedCacheNodesOfferingNotFoundFault) {
    ReservedCacheNodesOfferingNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodesOfferingNotFoundFault || (ReservedCacheNodesOfferingNotFoundFault = {}));
export var ServiceUpdateStatus;
(function (ServiceUpdateStatus) {
    ServiceUpdateStatus["AVAILABLE"] = "available";
    ServiceUpdateStatus["CANCELLED"] = "cancelled";
    ServiceUpdateStatus["EXPIRED"] = "expired";
})(ServiceUpdateStatus || (ServiceUpdateStatus = {}));
export var DescribeServiceUpdatesMessage;
(function (DescribeServiceUpdatesMessage) {
    DescribeServiceUpdatesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeServiceUpdatesMessage || (DescribeServiceUpdatesMessage = {}));
export var ServiceUpdateSeverity;
(function (ServiceUpdateSeverity) {
    ServiceUpdateSeverity["CRITICAL"] = "critical";
    ServiceUpdateSeverity["IMPORTANT"] = "important";
    ServiceUpdateSeverity["LOW"] = "low";
    ServiceUpdateSeverity["MEDIUM"] = "medium";
})(ServiceUpdateSeverity || (ServiceUpdateSeverity = {}));
export var ServiceUpdateType;
(function (ServiceUpdateType) {
    ServiceUpdateType["SECURITY_UPDATE"] = "security-update";
})(ServiceUpdateType || (ServiceUpdateType = {}));
export var ServiceUpdate;
(function (ServiceUpdate) {
    ServiceUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceUpdate || (ServiceUpdate = {}));
export var ServiceUpdatesMessage;
(function (ServiceUpdatesMessage) {
    ServiceUpdatesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceUpdatesMessage || (ServiceUpdatesMessage = {}));
export var DescribeSnapshotsListMessage;
(function (DescribeSnapshotsListMessage) {
    DescribeSnapshotsListMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSnapshotsListMessage || (DescribeSnapshotsListMessage = {}));
export var DescribeSnapshotsMessage;
(function (DescribeSnapshotsMessage) {
    DescribeSnapshotsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSnapshotsMessage || (DescribeSnapshotsMessage = {}));
export var TimeRangeFilter;
(function (TimeRangeFilter) {
    TimeRangeFilter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TimeRangeFilter || (TimeRangeFilter = {}));
export var DescribeUpdateActionsMessage;
(function (DescribeUpdateActionsMessage) {
    DescribeUpdateActionsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUpdateActionsMessage || (DescribeUpdateActionsMessage = {}));
export var NodeUpdateInitiatedBy;
(function (NodeUpdateInitiatedBy) {
    NodeUpdateInitiatedBy["CUSTOMER"] = "customer";
    NodeUpdateInitiatedBy["SYSTEM"] = "system";
})(NodeUpdateInitiatedBy || (NodeUpdateInitiatedBy = {}));
export var NodeUpdateStatus;
(function (NodeUpdateStatus) {
    NodeUpdateStatus["COMPLETE"] = "complete";
    NodeUpdateStatus["IN_PROGRESS"] = "in-progress";
    NodeUpdateStatus["NOT_APPLIED"] = "not-applied";
    NodeUpdateStatus["STOPPED"] = "stopped";
    NodeUpdateStatus["STOPPING"] = "stopping";
    NodeUpdateStatus["WAITING_TO_START"] = "waiting-to-start";
})(NodeUpdateStatus || (NodeUpdateStatus = {}));
export var CacheNodeUpdateStatus;
(function (CacheNodeUpdateStatus) {
    CacheNodeUpdateStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheNodeUpdateStatus || (CacheNodeUpdateStatus = {}));
export var NodeGroupMemberUpdateStatus;
(function (NodeGroupMemberUpdateStatus) {
    NodeGroupMemberUpdateStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupMemberUpdateStatus || (NodeGroupMemberUpdateStatus = {}));
export var NodeGroupUpdateStatus;
(function (NodeGroupUpdateStatus) {
    NodeGroupUpdateStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupUpdateStatus || (NodeGroupUpdateStatus = {}));
export var SlaMet;
(function (SlaMet) {
    SlaMet["NA"] = "n/a";
    SlaMet["NO"] = "no";
    SlaMet["YES"] = "yes";
})(SlaMet || (SlaMet = {}));
export var UpdateAction;
(function (UpdateAction) {
    UpdateAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAction || (UpdateAction = {}));
export var UpdateActionsMessage;
(function (UpdateActionsMessage) {
    UpdateActionsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateActionsMessage || (UpdateActionsMessage = {}));
export var DescribeUserGroupsMessage;
(function (DescribeUserGroupsMessage) {
    DescribeUserGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserGroupsMessage || (DescribeUserGroupsMessage = {}));
export var DescribeUserGroupsResult;
(function (DescribeUserGroupsResult) {
    DescribeUserGroupsResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserGroupsResult || (DescribeUserGroupsResult = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var DescribeUsersMessage;
(function (DescribeUsersMessage) {
    DescribeUsersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUsersMessage || (DescribeUsersMessage = {}));
export var DescribeUsersResult;
(function (DescribeUsersResult) {
    DescribeUsersResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUsersResult || (DescribeUsersResult = {}));
export var DisassociateGlobalReplicationGroupMessage;
(function (DisassociateGlobalReplicationGroupMessage) {
    DisassociateGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateGlobalReplicationGroupMessage || (DisassociateGlobalReplicationGroupMessage = {}));
export var DisassociateGlobalReplicationGroupResult;
(function (DisassociateGlobalReplicationGroupResult) {
    DisassociateGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateGlobalReplicationGroupResult || (DisassociateGlobalReplicationGroupResult = {}));
export var FailoverGlobalReplicationGroupMessage;
(function (FailoverGlobalReplicationGroupMessage) {
    FailoverGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailoverGlobalReplicationGroupMessage || (FailoverGlobalReplicationGroupMessage = {}));
export var FailoverGlobalReplicationGroupResult;
(function (FailoverGlobalReplicationGroupResult) {
    FailoverGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailoverGlobalReplicationGroupResult || (FailoverGlobalReplicationGroupResult = {}));
export var ReshardingConfiguration;
(function (ReshardingConfiguration) {
    ReshardingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReshardingConfiguration || (ReshardingConfiguration = {}));
export var RegionalConfiguration;
(function (RegionalConfiguration) {
    RegionalConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RegionalConfiguration || (RegionalConfiguration = {}));
export var IncreaseNodeGroupsInGlobalReplicationGroupMessage;
(function (IncreaseNodeGroupsInGlobalReplicationGroupMessage) {
    IncreaseNodeGroupsInGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IncreaseNodeGroupsInGlobalReplicationGroupMessage || (IncreaseNodeGroupsInGlobalReplicationGroupMessage = {}));
export var IncreaseNodeGroupsInGlobalReplicationGroupResult;
(function (IncreaseNodeGroupsInGlobalReplicationGroupResult) {
    IncreaseNodeGroupsInGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IncreaseNodeGroupsInGlobalReplicationGroupResult || (IncreaseNodeGroupsInGlobalReplicationGroupResult = {}));
export var IncreaseReplicaCountMessage;
(function (IncreaseReplicaCountMessage) {
    IncreaseReplicaCountMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IncreaseReplicaCountMessage || (IncreaseReplicaCountMessage = {}));
export var IncreaseReplicaCountResult;
(function (IncreaseReplicaCountResult) {
    IncreaseReplicaCountResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IncreaseReplicaCountResult || (IncreaseReplicaCountResult = {}));
export var InvalidKMSKeyFault;
(function (InvalidKMSKeyFault) {
    InvalidKMSKeyFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidKMSKeyFault || (InvalidKMSKeyFault = {}));
export var ListAllowedNodeTypeModificationsMessage;
(function (ListAllowedNodeTypeModificationsMessage) {
    ListAllowedNodeTypeModificationsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAllowedNodeTypeModificationsMessage || (ListAllowedNodeTypeModificationsMessage = {}));
export var ListTagsForResourceMessage;
(function (ListTagsForResourceMessage) {
    ListTagsForResourceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceMessage || (ListTagsForResourceMessage = {}));
export var ModifyCacheClusterMessage;
(function (ModifyCacheClusterMessage) {
    ModifyCacheClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyCacheClusterMessage || (ModifyCacheClusterMessage = {}));
export var ModifyCacheClusterResult;
(function (ModifyCacheClusterResult) {
    ModifyCacheClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyCacheClusterResult || (ModifyCacheClusterResult = {}));
export var CacheParameterGroupNameMessage;
(function (CacheParameterGroupNameMessage) {
    CacheParameterGroupNameMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CacheParameterGroupNameMessage || (CacheParameterGroupNameMessage = {}));
export var ParameterNameValue;
(function (ParameterNameValue) {
    ParameterNameValue.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ParameterNameValue || (ParameterNameValue = {}));
export var ModifyCacheParameterGroupMessage;
(function (ModifyCacheParameterGroupMessage) {
    ModifyCacheParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyCacheParameterGroupMessage || (ModifyCacheParameterGroupMessage = {}));
export var ModifyCacheSubnetGroupMessage;
(function (ModifyCacheSubnetGroupMessage) {
    ModifyCacheSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyCacheSubnetGroupMessage || (ModifyCacheSubnetGroupMessage = {}));
export var ModifyCacheSubnetGroupResult;
(function (ModifyCacheSubnetGroupResult) {
    ModifyCacheSubnetGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyCacheSubnetGroupResult || (ModifyCacheSubnetGroupResult = {}));
export var SubnetInUse;
(function (SubnetInUse) {
    SubnetInUse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubnetInUse || (SubnetInUse = {}));
export var ModifyGlobalReplicationGroupMessage;
(function (ModifyGlobalReplicationGroupMessage) {
    ModifyGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyGlobalReplicationGroupMessage || (ModifyGlobalReplicationGroupMessage = {}));
export var ModifyGlobalReplicationGroupResult;
(function (ModifyGlobalReplicationGroupResult) {
    ModifyGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyGlobalReplicationGroupResult || (ModifyGlobalReplicationGroupResult = {}));
export var ModifyReplicationGroupMessage;
(function (ModifyReplicationGroupMessage) {
    ModifyReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyReplicationGroupMessage || (ModifyReplicationGroupMessage = {}));
export var ModifyReplicationGroupResult;
(function (ModifyReplicationGroupResult) {
    ModifyReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyReplicationGroupResult || (ModifyReplicationGroupResult = {}));
export var ModifyReplicationGroupShardConfigurationMessage;
(function (ModifyReplicationGroupShardConfigurationMessage) {
    ModifyReplicationGroupShardConfigurationMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyReplicationGroupShardConfigurationMessage || (ModifyReplicationGroupShardConfigurationMessage = {}));
export var ModifyReplicationGroupShardConfigurationResult;
(function (ModifyReplicationGroupShardConfigurationResult) {
    ModifyReplicationGroupShardConfigurationResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyReplicationGroupShardConfigurationResult || (ModifyReplicationGroupShardConfigurationResult = {}));
export var ModifyUserMessage;
(function (ModifyUserMessage) {
    ModifyUserMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyUserMessage || (ModifyUserMessage = {}));
export var ModifyUserGroupMessage;
(function (ModifyUserGroupMessage) {
    ModifyUserGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyUserGroupMessage || (ModifyUserGroupMessage = {}));
export var PurchaseReservedCacheNodesOfferingMessage;
(function (PurchaseReservedCacheNodesOfferingMessage) {
    PurchaseReservedCacheNodesOfferingMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedCacheNodesOfferingMessage || (PurchaseReservedCacheNodesOfferingMessage = {}));
export var PurchaseReservedCacheNodesOfferingResult;
(function (PurchaseReservedCacheNodesOfferingResult) {
    PurchaseReservedCacheNodesOfferingResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedCacheNodesOfferingResult || (PurchaseReservedCacheNodesOfferingResult = {}));
export var ReservedCacheNodeAlreadyExistsFault;
(function (ReservedCacheNodeAlreadyExistsFault) {
    ReservedCacheNodeAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodeAlreadyExistsFault || (ReservedCacheNodeAlreadyExistsFault = {}));
export var ReservedCacheNodeQuotaExceededFault;
(function (ReservedCacheNodeQuotaExceededFault) {
    ReservedCacheNodeQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedCacheNodeQuotaExceededFault || (ReservedCacheNodeQuotaExceededFault = {}));
export var RebalanceSlotsInGlobalReplicationGroupMessage;
(function (RebalanceSlotsInGlobalReplicationGroupMessage) {
    RebalanceSlotsInGlobalReplicationGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebalanceSlotsInGlobalReplicationGroupMessage || (RebalanceSlotsInGlobalReplicationGroupMessage = {}));
export var RebalanceSlotsInGlobalReplicationGroupResult;
(function (RebalanceSlotsInGlobalReplicationGroupResult) {
    RebalanceSlotsInGlobalReplicationGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebalanceSlotsInGlobalReplicationGroupResult || (RebalanceSlotsInGlobalReplicationGroupResult = {}));
export var RebootCacheClusterMessage;
(function (RebootCacheClusterMessage) {
    RebootCacheClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebootCacheClusterMessage || (RebootCacheClusterMessage = {}));
export var RebootCacheClusterResult;
(function (RebootCacheClusterResult) {
    RebootCacheClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RebootCacheClusterResult || (RebootCacheClusterResult = {}));
export var RemoveTagsFromResourceMessage;
(function (RemoveTagsFromResourceMessage) {
    RemoveTagsFromResourceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTagsFromResourceMessage || (RemoveTagsFromResourceMessage = {}));
export var TagNotFoundFault;
(function (TagNotFoundFault) {
    TagNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagNotFoundFault || (TagNotFoundFault = {}));
export var ResetCacheParameterGroupMessage;
(function (ResetCacheParameterGroupMessage) {
    ResetCacheParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResetCacheParameterGroupMessage || (ResetCacheParameterGroupMessage = {}));
export var AuthorizationNotFoundFault;
(function (AuthorizationNotFoundFault) {
    AuthorizationNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizationNotFoundFault || (AuthorizationNotFoundFault = {}));
export var RevokeCacheSecurityGroupIngressMessage;
(function (RevokeCacheSecurityGroupIngressMessage) {
    RevokeCacheSecurityGroupIngressMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeCacheSecurityGroupIngressMessage || (RevokeCacheSecurityGroupIngressMessage = {}));
export var RevokeCacheSecurityGroupIngressResult;
(function (RevokeCacheSecurityGroupIngressResult) {
    RevokeCacheSecurityGroupIngressResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeCacheSecurityGroupIngressResult || (RevokeCacheSecurityGroupIngressResult = {}));
export var ReplicationGroupAlreadyUnderMigrationFault;
(function (ReplicationGroupAlreadyUnderMigrationFault) {
    ReplicationGroupAlreadyUnderMigrationFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupAlreadyUnderMigrationFault || (ReplicationGroupAlreadyUnderMigrationFault = {}));
export var CustomerNodeEndpoint;
(function (CustomerNodeEndpoint) {
    CustomerNodeEndpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomerNodeEndpoint || (CustomerNodeEndpoint = {}));
export var StartMigrationMessage;
(function (StartMigrationMessage) {
    StartMigrationMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartMigrationMessage || (StartMigrationMessage = {}));
export var StartMigrationResponse;
(function (StartMigrationResponse) {
    StartMigrationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartMigrationResponse || (StartMigrationResponse = {}));
export var APICallRateForCustomerExceededFault;
(function (APICallRateForCustomerExceededFault) {
    APICallRateForCustomerExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(APICallRateForCustomerExceededFault || (APICallRateForCustomerExceededFault = {}));
export var NodeGroupNotFoundFault;
(function (NodeGroupNotFoundFault) {
    NodeGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeGroupNotFoundFault || (NodeGroupNotFoundFault = {}));
export var TestFailoverMessage;
(function (TestFailoverMessage) {
    TestFailoverMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TestFailoverMessage || (TestFailoverMessage = {}));
export var TestFailoverNotAvailableFault;
(function (TestFailoverNotAvailableFault) {
    TestFailoverNotAvailableFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TestFailoverNotAvailableFault || (TestFailoverNotAvailableFault = {}));
export var TestFailoverResult;
(function (TestFailoverResult) {
    TestFailoverResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TestFailoverResult || (TestFailoverResult = {}));
