// @aws-sdk/client-elasticache/dist-es/commands/AuthorizeCacheSecurityGroupIngressCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AuthorizeCacheSecurityGroupIngressMessage, AuthorizeCacheSecurityGroupIngressResult, } from "../models/models_0";
import { deserializeAws_queryAuthorizeCacheSecurityGroupIngressCommand, serializeAws_queryAuthorizeCacheSecurityGroupIngressCommand, } from "../protocols/Aws_query";
var AuthorizeCacheSecurityGroupIngressCommand = (function (_super) {
    __extends(AuthorizeCacheSecurityGroupIngressCommand, _super);
    function AuthorizeCacheSecurityGroupIngressCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AuthorizeCacheSecurityGroupIngressCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "AuthorizeCacheSecurityGroupIngressCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AuthorizeCacheSecurityGroupIngressMessage.filterSensitiveLog,
            outputFilterSensitiveLog: AuthorizeCacheSecurityGroupIngressResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AuthorizeCacheSecurityGroupIngressCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryAuthorizeCacheSecurityGroupIngressCommand(input, context);
    };
    AuthorizeCacheSecurityGroupIngressCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryAuthorizeCacheSecurityGroupIngressCommand(output, context);
    };
    return AuthorizeCacheSecurityGroupIngressCommand;
}($Command));
export { AuthorizeCacheSecurityGroupIngressCommand };
