// @aws-sdk/client-elasticache/dist-es/commands/CompleteMigrationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CompleteMigrationMessage, CompleteMigrationResponse } from "../models/models_0";
import { deserializeAws_queryCompleteMigrationCommand, serializeAws_queryCompleteMigrationCommand, } from "../protocols/Aws_query";
var CompleteMigrationCommand = (function (_super) {
    __extends(CompleteMigrationCommand, _super);
    function CompleteMigrationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CompleteMigrationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "CompleteMigrationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CompleteMigrationMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CompleteMigrationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CompleteMigrationCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCompleteMigrationCommand(input, context);
    };
    CompleteMigrationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCompleteMigrationCommand(output, context);
    };
    return CompleteMigrationCommand;
}($Command));
export { CompleteMigrationCommand };
