// @aws-sdk/client-elasticache/dist-es/commands/CreateReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateReplicationGroupMessage, CreateReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateReplicationGroupCommand, serializeAws_queryCreateReplicationGroupCommand, } from "../protocols/Aws_query";
var CreateReplicationGroupCommand = (function (_super) {
    __extends(CreateReplicationGroupCommand, _super);
    function CreateReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "CreateReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateReplicationGroupCommand(input, context);
    };
    CreateReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateReplicationGroupCommand(output, context);
    };
    return CreateReplicationGroupCommand;
}($Command));
export { CreateReplicationGroupCommand };
