// @aws-sdk/client-elasticache/dist-es/commands/CreateCacheSubnetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCacheSubnetGroupMessage, CreateCacheSubnetGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateCacheSubnetGroupCommand, serializeAws_queryCreateCacheSubnetGroupCommand, } from "../protocols/Aws_query";
var CreateCacheSubnetGroupCommand = (function (_super) {
    __extends(CreateCacheSubnetGroupCommand, _super);
    function CreateCacheSubnetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCacheSubnetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "CreateCacheSubnetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCacheSubnetGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCacheSubnetGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCacheSubnetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateCacheSubnetGroupCommand(input, context);
    };
    CreateCacheSubnetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateCacheSubnetGroupCommand(output, context);
    };
    return CreateCacheSubnetGroupCommand;
}($Command));
export { CreateCacheSubnetGroupCommand };
