// @aws-sdk/client-elasticache/dist-es/commands/CreateCacheClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCacheClusterMessage, CreateCacheClusterResult } from "../models/models_0";
import { deserializeAws_queryCreateCacheClusterCommand, serializeAws_queryCreateCacheClusterCommand, } from "../protocols/Aws_query";
var CreateCacheClusterCommand = (function (_super) {
    __extends(CreateCacheClusterCommand, _super);
    function CreateCacheClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCacheClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "CreateCacheClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCacheClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCacheClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCacheClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateCacheClusterCommand(input, context);
    };
    CreateCacheClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateCacheClusterCommand(output, context);
    };
    return CreateCacheClusterCommand;
}($Command));
export { CreateCacheClusterCommand };
