// @aws-sdk/client-elasticache/dist-es/commands/CreateCacheSecurityGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCacheSecurityGroupMessage, CreateCacheSecurityGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateCacheSecurityGroupCommand, serializeAws_queryCreateCacheSecurityGroupCommand, } from "../protocols/Aws_query";
var CreateCacheSecurityGroupCommand = (function (_super) {
    __extends(CreateCacheSecurityGroupCommand, _super);
    function CreateCacheSecurityGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCacheSecurityGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "CreateCacheSecurityGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCacheSecurityGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCacheSecurityGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCacheSecurityGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateCacheSecurityGroupCommand(input, context);
    };
    CreateCacheSecurityGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateCacheSecurityGroupCommand(output, context);
    };
    return CreateCacheSecurityGroupCommand;
}($Command));
export { CreateCacheSecurityGroupCommand };
