// @aws-sdk/client-elasticache/dist-es/commands/DecreaseReplicaCountCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DecreaseReplicaCountMessage, DecreaseReplicaCountResult } from "../models/models_0";
import { deserializeAws_queryDecreaseReplicaCountCommand, serializeAws_queryDecreaseReplicaCountCommand, } from "../protocols/Aws_query";
var DecreaseReplicaCountCommand = (function (_super) {
    __extends(DecreaseReplicaCountCommand, _super);
    function DecreaseReplicaCountCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DecreaseReplicaCountCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DecreaseReplicaCountCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DecreaseReplicaCountMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DecreaseReplicaCountResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DecreaseReplicaCountCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDecreaseReplicaCountCommand(input, context);
    };
    DecreaseReplicaCountCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDecreaseReplicaCountCommand(output, context);
    };
    return DecreaseReplicaCountCommand;
}($Command));
export { DecreaseReplicaCountCommand };
