// @aws-sdk/client-elasticache/dist-es/commands/DeleteGlobalReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteGlobalReplicationGroupMessage, DeleteGlobalReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryDeleteGlobalReplicationGroupCommand, serializeAws_queryDeleteGlobalReplicationGroupCommand, } from "../protocols/Aws_query";
var DeleteGlobalReplicationGroupCommand = (function (_super) {
    __extends(DeleteGlobalReplicationGroupCommand, _super);
    function DeleteGlobalReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteGlobalReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DeleteGlobalReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteGlobalReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteGlobalReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteGlobalReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteGlobalReplicationGroupCommand(input, context);
    };
    DeleteGlobalReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteGlobalReplicationGroupCommand(output, context);
    };
    return DeleteGlobalReplicationGroupCommand;
}($Command));
export { DeleteGlobalReplicationGroupCommand };
