// @aws-sdk/client-elasticache/dist-es/commands/DeleteReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteReplicationGroupMessage, DeleteReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryDeleteReplicationGroupCommand, serializeAws_queryDeleteReplicationGroupCommand, } from "../protocols/Aws_query";
var DeleteReplicationGroupCommand = (function (_super) {
    __extends(DeleteReplicationGroupCommand, _super);
    function DeleteReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DeleteReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteReplicationGroupCommand(input, context);
    };
    DeleteReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteReplicationGroupCommand(output, context);
    };
    return DeleteReplicationGroupCommand;
}($Command));
export { DeleteReplicationGroupCommand };
