// @aws-sdk/client-elasticache/dist-es/commands/DescribeReservedCacheNodesOfferingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReservedCacheNodesOfferingsMessage, ReservedCacheNodesOfferingMessage } from "../models/models_0";
import { deserializeAws_queryDescribeReservedCacheNodesOfferingsCommand, serializeAws_queryDescribeReservedCacheNodesOfferingsCommand, } from "../protocols/Aws_query";
var DescribeReservedCacheNodesOfferingsCommand = (function (_super) {
    __extends(DescribeReservedCacheNodesOfferingsCommand, _super);
    function DescribeReservedCacheNodesOfferingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReservedCacheNodesOfferingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DescribeReservedCacheNodesOfferingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReservedCacheNodesOfferingsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ReservedCacheNodesOfferingMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReservedCacheNodesOfferingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeReservedCacheNodesOfferingsCommand(input, context);
    };
    DescribeReservedCacheNodesOfferingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeReservedCacheNodesOfferingsCommand(output, context);
    };
    return DescribeReservedCacheNodesOfferingsCommand;
}($Command));
export { DescribeReservedCacheNodesOfferingsCommand };
