// @aws-sdk/client-elasticache/dist-es/commands/DescribeUpdateActionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUpdateActionsMessage, UpdateActionsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeUpdateActionsCommand, serializeAws_queryDescribeUpdateActionsCommand, } from "../protocols/Aws_query";
var DescribeUpdateActionsCommand = (function (_super) {
    __extends(DescribeUpdateActionsCommand, _super);
    function DescribeUpdateActionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUpdateActionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DescribeUpdateActionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUpdateActionsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateActionsMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUpdateActionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeUpdateActionsCommand(input, context);
    };
    DescribeUpdateActionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeUpdateActionsCommand(output, context);
    };
    return DescribeUpdateActionsCommand;
}($Command));
export { DescribeUpdateActionsCommand };
