// @aws-sdk/client-elasticache/dist-es/commands/DescribeCacheEngineVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CacheEngineVersionMessage, DescribeCacheEngineVersionsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeCacheEngineVersionsCommand, serializeAws_queryDescribeCacheEngineVersionsCommand, } from "../protocols/Aws_query";
var DescribeCacheEngineVersionsCommand = (function (_super) {
    __extends(DescribeCacheEngineVersionsCommand, _super);
    function DescribeCacheEngineVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCacheEngineVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DescribeCacheEngineVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCacheEngineVersionsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CacheEngineVersionMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCacheEngineVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeCacheEngineVersionsCommand(input, context);
    };
    DescribeCacheEngineVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeCacheEngineVersionsCommand(output, context);
    };
    return DescribeCacheEngineVersionsCommand;
}($Command));
export { DescribeCacheEngineVersionsCommand };
