// @aws-sdk/client-elasticache/dist-es/commands/DescribeCacheSubnetGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CacheSubnetGroupMessage, DescribeCacheSubnetGroupsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeCacheSubnetGroupsCommand, serializeAws_queryDescribeCacheSubnetGroupsCommand, } from "../protocols/Aws_query";
var DescribeCacheSubnetGroupsCommand = (function (_super) {
    __extends(DescribeCacheSubnetGroupsCommand, _super);
    function DescribeCacheSubnetGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCacheSubnetGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "DescribeCacheSubnetGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCacheSubnetGroupsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CacheSubnetGroupMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCacheSubnetGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeCacheSubnetGroupsCommand(input, context);
    };
    DescribeCacheSubnetGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeCacheSubnetGroupsCommand(output, context);
    };
    return DescribeCacheSubnetGroupsCommand;
}($Command));
export { DescribeCacheSubnetGroupsCommand };
