// @aws-sdk/client-elasticache/dist-es/commands/FailoverGlobalReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FailoverGlobalReplicationGroupMessage, FailoverGlobalReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryFailoverGlobalReplicationGroupCommand, serializeAws_queryFailoverGlobalReplicationGroupCommand, } from "../protocols/Aws_query";
var FailoverGlobalReplicationGroupCommand = (function (_super) {
    __extends(FailoverGlobalReplicationGroupCommand, _super);
    function FailoverGlobalReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    FailoverGlobalReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "FailoverGlobalReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: FailoverGlobalReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: FailoverGlobalReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    FailoverGlobalReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryFailoverGlobalReplicationGroupCommand(input, context);
    };
    FailoverGlobalReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryFailoverGlobalReplicationGroupCommand(output, context);
    };
    return FailoverGlobalReplicationGroupCommand;
}($Command));
export { FailoverGlobalReplicationGroupCommand };
