// @aws-sdk/client-elasticache/dist-es/commands/ListAllowedNodeTypeModificationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AllowedNodeTypeModificationsMessage, ListAllowedNodeTypeModificationsMessage } from "../models/models_0";
import { deserializeAws_queryListAllowedNodeTypeModificationsCommand, serializeAws_queryListAllowedNodeTypeModificationsCommand, } from "../protocols/Aws_query";
var ListAllowedNodeTypeModificationsCommand = (function (_super) {
    __extends(ListAllowedNodeTypeModificationsCommand, _super);
    function ListAllowedNodeTypeModificationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListAllowedNodeTypeModificationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ListAllowedNodeTypeModificationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListAllowedNodeTypeModificationsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: AllowedNodeTypeModificationsMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListAllowedNodeTypeModificationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListAllowedNodeTypeModificationsCommand(input, context);
    };
    ListAllowedNodeTypeModificationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListAllowedNodeTypeModificationsCommand(output, context);
    };
    return ListAllowedNodeTypeModificationsCommand;
}($Command));
export { ListAllowedNodeTypeModificationsCommand };
