// @aws-sdk/client-elasticache/dist-es/commands/ModifyReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyReplicationGroupMessage, ModifyReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryModifyReplicationGroupCommand, serializeAws_queryModifyReplicationGroupCommand, } from "../protocols/Aws_query";
var ModifyReplicationGroupCommand = (function (_super) {
    __extends(ModifyReplicationGroupCommand, _super);
    function ModifyReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ModifyReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyReplicationGroupCommand(input, context);
    };
    ModifyReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyReplicationGroupCommand(output, context);
    };
    return ModifyReplicationGroupCommand;
}($Command));
export { ModifyReplicationGroupCommand };
