// @aws-sdk/client-elasticache/dist-es/commands/ModifyReplicationGroupShardConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyReplicationGroupShardConfigurationMessage, ModifyReplicationGroupShardConfigurationResult, } from "../models/models_0";
import { deserializeAws_queryModifyReplicationGroupShardConfigurationCommand, serializeAws_queryModifyReplicationGroupShardConfigurationCommand, } from "../protocols/Aws_query";
var ModifyReplicationGroupShardConfigurationCommand = (function (_super) {
    __extends(ModifyReplicationGroupShardConfigurationCommand, _super);
    function ModifyReplicationGroupShardConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyReplicationGroupShardConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ModifyReplicationGroupShardConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyReplicationGroupShardConfigurationMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyReplicationGroupShardConfigurationResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyReplicationGroupShardConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyReplicationGroupShardConfigurationCommand(input, context);
    };
    ModifyReplicationGroupShardConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyReplicationGroupShardConfigurationCommand(output, context);
    };
    return ModifyReplicationGroupShardConfigurationCommand;
}($Command));
export { ModifyReplicationGroupShardConfigurationCommand };
