// @aws-sdk/client-elasticache/dist-es/commands/ModifyUserGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyUserGroupMessage, UserGroup } from "../models/models_0";
import { deserializeAws_queryModifyUserGroupCommand, serializeAws_queryModifyUserGroupCommand, } from "../protocols/Aws_query";
var ModifyUserGroupCommand = (function (_super) {
    __extends(ModifyUserGroupCommand, _super);
    function ModifyUserGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyUserGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ModifyUserGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyUserGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: UserGroup.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyUserGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyUserGroupCommand(input, context);
    };
    ModifyUserGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyUserGroupCommand(output, context);
    };
    return ModifyUserGroupCommand;
}($Command));
export { ModifyUserGroupCommand };
