// @aws-sdk/client-elasticache/dist-es/commands/ModifyGlobalReplicationGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyGlobalReplicationGroupMessage, ModifyGlobalReplicationGroupResult } from "../models/models_0";
import { deserializeAws_queryModifyGlobalReplicationGroupCommand, serializeAws_queryModifyGlobalReplicationGroupCommand, } from "../protocols/Aws_query";
var ModifyGlobalReplicationGroupCommand = (function (_super) {
    __extends(ModifyGlobalReplicationGroupCommand, _super);
    function ModifyGlobalReplicationGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyGlobalReplicationGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ModifyGlobalReplicationGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyGlobalReplicationGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyGlobalReplicationGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyGlobalReplicationGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyGlobalReplicationGroupCommand(input, context);
    };
    ModifyGlobalReplicationGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyGlobalReplicationGroupCommand(output, context);
    };
    return ModifyGlobalReplicationGroupCommand;
}($Command));
export { ModifyGlobalReplicationGroupCommand };
