// @aws-sdk/client-elasticache/dist-es/commands/ModifyCacheParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CacheParameterGroupNameMessage, ModifyCacheParameterGroupMessage } from "../models/models_0";
import { deserializeAws_queryModifyCacheParameterGroupCommand, serializeAws_queryModifyCacheParameterGroupCommand, } from "../protocols/Aws_query";
var ModifyCacheParameterGroupCommand = (function (_super) {
    __extends(ModifyCacheParameterGroupCommand, _super);
    function ModifyCacheParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyCacheParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ModifyCacheParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyCacheParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CacheParameterGroupNameMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyCacheParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyCacheParameterGroupCommand(input, context);
    };
    ModifyCacheParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyCacheParameterGroupCommand(output, context);
    };
    return ModifyCacheParameterGroupCommand;
}($Command));
export { ModifyCacheParameterGroupCommand };
