// @aws-sdk/client-elasticache/dist-es/commands/ResetCacheParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CacheParameterGroupNameMessage, ResetCacheParameterGroupMessage } from "../models/models_0";
import { deserializeAws_queryResetCacheParameterGroupCommand, serializeAws_queryResetCacheParameterGroupCommand, } from "../protocols/Aws_query";
var ResetCacheParameterGroupCommand = (function (_super) {
    __extends(ResetCacheParameterGroupCommand, _super);
    function ResetCacheParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResetCacheParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "ResetCacheParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResetCacheParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CacheParameterGroupNameMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResetCacheParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryResetCacheParameterGroupCommand(input, context);
    };
    ResetCacheParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryResetCacheParameterGroupCommand(output, context);
    };
    return ResetCacheParameterGroupCommand;
}($Command));
export { ResetCacheParameterGroupCommand };
