// @aws-sdk/client-elasticache/dist-es/commands/RevokeCacheSecurityGroupIngressCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RevokeCacheSecurityGroupIngressMessage, RevokeCacheSecurityGroupIngressResult } from "../models/models_0";
import { deserializeAws_queryRevokeCacheSecurityGroupIngressCommand, serializeAws_queryRevokeCacheSecurityGroupIngressCommand, } from "../protocols/Aws_query";
var RevokeCacheSecurityGroupIngressCommand = (function (_super) {
    __extends(RevokeCacheSecurityGroupIngressCommand, _super);
    function RevokeCacheSecurityGroupIngressCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RevokeCacheSecurityGroupIngressCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElastiCacheClient";
        var commandName = "RevokeCacheSecurityGroupIngressCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RevokeCacheSecurityGroupIngressMessage.filterSensitiveLog,
            outputFilterSensitiveLog: RevokeCacheSecurityGroupIngressResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RevokeCacheSecurityGroupIngressCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRevokeCacheSecurityGroupIngressCommand(input, context);
    };
    RevokeCacheSecurityGroupIngressCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRevokeCacheSecurityGroupIngressCommand(output, context);
    };
    return RevokeCacheSecurityGroupIngressCommand;
}($Command));
export { RevokeCacheSecurityGroupIngressCommand };
